config/m4/cuda.m4

```diff
@@ -51,10 +51,24 @@ AS_IF([test "x$cuda_checked" != "xyes"],
          AS_IF([test "x$cuda_happy" = "xyes"],
                [AC_CHECK_LIB([cudart], [cudaGetDeviceCount],
                              [CUDA_LIBS="$CUDA_LIBS -lcudart"], [cuda_happy="no"])])
+
+         # Check nvml header files
+         AC_CHECK_HEADERS([nvml.h],
+                          [cuda_happy="yes"], [cuda_happy="no"])
+
+         AS_IF([test "x$cuda_happy" = "xyes"],
+               [],
+               [AC_MSG_ERROR([nvml header not found. Install appropriate cuda-nvml-devel-X-Y package])])
+
+         # Check nvml library
          AS_IF([test "x$cuda_happy" = "xyes"],
                [AC_CHECK_LIB([nvidia-ml], [nvmlInit],
                              [CUDA_LIBS="$CUDA_LIBS -lnvidia-ml"], [cuda_happy="no"])])
 
+         AS_IF([test "x$cuda_happy" = "xyes"],
```

Comment:
1. Pls do it like this:
 https://github.com/openucx/ucx/blob/29aa1f2abdb39a6d32f8b00fd2a65e2321139e70/src/ucs/configure.m4#L121
2. err message is:
`libnvidia-ml.so is not found. Please install appropriate nvidia driver package`
3. can add it as the "else" part of the AC_CHECK_LIB in line 65 (`test x$with_cuda != -xguess`)